internal/decryption/encrypted_audit_log.go

```diff
@@ -71,11 +71,17 @@ func (logLine *AuditLogLine) KeyProvider(opts *KeyProviderOpts) (keyproviders.Ke
 
 	switch *logLine.KeyStoreIdentifier.Provider {
 	case keyproviders.LocalKey:
+		if opts.Local == nil {
+			return nil, fmt.Errorf("keyProvider %s not supported", *logLine.KeyStoreIdentifier.Provider)
```

Comment:
[style] this and others like it, consider a sentinel error 
```go
var ErrInvalidKeyProvider = errors.New("unsupported key provider")
...
fmt.Errorf("%w: %s", ErrInvalidKeyProvider, *logLine.KeyStoreIdentifier.Provider)
```
this has the advantage later on we can do `errors.Is(err, ErrInvalidKeyProvider)` and catch it for all